docs/plugins.md

```diff
@@ -0,0 +1,352 @@
+
+**Example `package.json`**
+
+```json
+{
+  "name": "lighthouse-plugin-cats",
+  "main": "plugin.js",
+  "peerDependencies": {
+    "lighthouse": "^5.0.0"
+  }
+}
+```
+
+#### `plugin.js`
+
+This file contains the configuration for your plugin. It can be called anything you like, just ensure it is referenced by the `"main"` property in your `package.json`.
+
+**Example `plugin.js`**
+
+```js
+module.exports = {
+  // Additional audit to run on information Lighthouse gathered.
+  audits: [{path: 'lighthouse-plugin-cats/cat-audit.js'}],
+
+  // A new category in the report for the plugin output.
+  category: {
+    title: 'Cats',
+    description: 'Results for our new cat plugin category.',
+    auditRefs: [{id: 'cat-audit-id', weight: 1}],
+  },
+};
+```
+
+#### Custom Audit
+
+These files contain the logic that will generate results for the Lighthouse report. An audit is a class with two important properties:
+
+1. `meta` - This contains important information about how the audit will be referenced in the config and how it will be displayed in the HTML report.
+2. `audit` - This is a function that should return the audit's results.
+
+**Example `cat-audit.js`**
+
+```js
+const Audit = require('lighthouse').Audit;
+
+class CatAudit extends Audit {
+  static get meta() {
+    return {
+      id: 'cat-audit-id',
+      title: 'Page has least one cat image',
+      failureTitle: 'Page does not have at least one cat image',
+      description: 'Pages should have lots of cat images to keep users happy. ' +
+        'Consider adding a picture of a cat to your page improve engagement.',
+      requiredArtifacts: ['ImageElements'],
+    };
+  }
+
+  static audit(artifacts) {
+    // Artifacts requested in `requiredArtifacts` above are passed to your audit.
+    // See the "API -> Plugin Audits" section below for what artifacts are available.
+    const images = artifacts.ImageElements;
+    const catImages = images.filter(image => image.src.toLowerCase().includes('cat'));
+
+    return {
+      // Give users a 100 if they had a cat image, 0 if they didn't.
+      score: catImages.length > 0 ? 1 : 0,
+      // Also return the total number of cat images that can be used by report JSON consumers.
+      numericValue: catImages.length,
+    }
+  }
+}
+
+module.exports = CatAudit;
+```
+
+## API
+
+### Plugin Config
+
+The plugin config file (`plugin.js` in the example and recipe above) is a subset of the available [configuration](./configuration.md) for full custom Lighthouse config files.
+
+A plugin config is an object that has at least two properties: `audits` and `category`.
+
+#### `audits`
+
+Defines the new audits the plugin adds. It is an array of string paths to the audit files. Each path should be treated as an absolute string a user of your module might pass to `require`, so use paths of the form `lighthouse-plugin-<your plugin>/path/to/audit.js`.
```

Comment:
`lighthouse-plugin-<your plugin>/path/to/audit.js`.

to

`lighthouse-plugin-<your plugin>/path/to/audits/audit-file.js`.